### PR TITLE
nvm plugin improvements

### LINKED ISSUE #5270: nvm functionality doesn't work if nvm is installed in a custom directory
nvm supports being installed in a custom directory, by default it's installed in ~/.nvm but it can be specified anywhere by using the environment variable `$NVM_DIR`. Even if nvm is installed in `~/.nvm` the environment variable should still be set, otherwise nvm has to work out where it's installed each time you run it.

Currently all omz functionality regarding zsh (loader/completion/theme helper) hardcode the path as ~/.nvm and `$NVM_DIR` is never set.

### PR DESCRIPTION
**Pull request by lukechilds**
`nvm` isn't always installed in ~/.nvm. These changes allows nvm to be installed in any directory. It also now properly exports the `$NVM_DIR` env var if it hasn't been set and checks it in the completion and `nvm_prompt_info` rather than assuming `nvm` is in ~/.nvm.

Fixes #5270

### CODE REVIEW

**Comment by mcornella** [comment]
:+1: but I would put the `$NVM_DIR` definition on `lib/nvm.zsh`, to have it as soon as possible.

**Comment by lukechilds** [comment]
@mcornella Ah, I assumed `plugins/nvm/nvm.plugin.zsh` was loaded before `lib/nvm.zsh` (not sure why looking back 😃 🔫 ).

I'm not too familiar with omz, is there ever a scenario where `plugins/nvm/nvm.plugin.zsh`would be sourced but not `lib/nvm.zsh`? I'm  thinking it may be best to put the `$NVM_DIR` definition in both to make sure it's definitely always set.

**Comment by lukechilds** [comment]
Wait, isn't `lib/nvm.zsh` setting up `nvm_prompt_info` as a helper function for a theme? So the theme helper function will always be called after the `nvm` plugin has been loaded so `$NVM_DIR` will exist?

**Comment by lukechilds** [comment]
@mcornella Is this ok as is?

**Comment by mcornella** [comment]
I wasn't sure what would happen if `$NVM_DIR` wasn't set, but after revising it further I see no possible problem with `nvm_prompt_info`. The exporting of `NVM_DIR` is [already done on `nvm.sh`](https://github.com/creationix/nvm/blob/364096a/nvm.sh#L113-L114) but it doesn't hurt anything doing it here.

TL;DR this is already ok.

**Comment by hax** [comment]
Should lib/nvm.zsh move to /plugins/nvm ?
